src/include/platform/internal/GenericPlatformManagerImpl_FreeRTOS.ipp

```diff
@@ -275,6 +275,17 @@ void GenericPlatformManagerImpl_FreeRTOS<ImplClass>::PostEventFromISR(const Chip
 template <class ImplClass>
 void GenericPlatformManagerImpl_FreeRTOS<ImplClass>::_Shutdown(void)
 {
+    GenericPlatformManagerImpl<ImplClass>::_Shutdown();
+
+    if (mChipEventQueue != NULL) {
+        vQueueDelete(mChipEventQueue);
```

Comment:
Do we need to delete them? Not sure how the init is done but if `mChipEventQueue` isn't null we wouldn't have to create a new one.

Reply:
> Do we need to delete them?

We could avoid that, I think, if Init checked for a non-null one and skipped allocating a new one.  I would be fine with that; just what we have right now looks like it will leak if you init/shutdown/init.

Reply:
Done.